NetCasbin/InternalEnforcer.cs

```diff
@@ -17,34 +17,29 @@ public class InternalEnforcer : CoreEnforcer
         /// <returns></returns>
         protected bool AddPolicy(string sec, string ptype, List<string> rule)
         {
-            var ruleAdded = model.AddPolicy(sec, ptype, rule);
-            if (!ruleAdded)
+            if (model.HasPolicy(sec, ptype, rule))
             {
-                return false;
-            } 
-
+                return true;
+            }
+            
             if (adapter != null && autoSave)
             {
                 try
                 {
                     adapter.AddPolicy(sec, ptype, rule);
                 }
-                catch (NotImplementedException e)
+                catch (NotImplementedException)
                 {
-
+                    // error intentionally ignored
                 }
-                catch (Exception e)
+                finally
                 {
-                    throw e;
-                }
-
-                if (watcher != null)
-                {
-                    // error intentionally ignored
-                    watcher.Update();
+                    watcher?.Update();
```

Comment:
If `NotImplementedException` is caught, it means the adapter doesn't support Auto-Save, the adapter side will always fail. DB is also not updated. Why would we update the watcher?